File: Models/Competidor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Prisma.Models
{
    [Table("Competidores")]
    public class Competidor
    {
        [Key]
        public int Id {get;set;}
        [Required]
        public string Codigo {get;set;}
        [Required]
        public string Nombre {get;set;}
        public string Calle {get;set;}
        public decimal Latitud {get;set;}
        public decimal Longitud {get;set;}
        public bool Marcador {get;set;}
        public bool Observar {get;set;}
        [Required]
        public int MarcaId { get; set; }
        public Marca Marca {get;set;}
        [Required]
        public int ZonaPrecioId { get; set; }
        public ZonaPrecio ZonaPrecio {get;set;}
    }
}

File: ClientApp/src/app/models/competidor.ts
export class Competidor {
    constructor(
        public id :number,
        public codigo :string,
        public nombre :string,
        public calle :string,
        public latitud :number,
        public longitud :number,
        public marcador :boolean,
        public observar :boolean,
        public marcaId :number,
        public zonaPrecioId :number
        )
        {}
}

export class CompetidorList {
    constructor(
        public id :number,
        public codigo :string,
        public nombre :string,
        public calle :string,
        public latitud :number,
        public longitud :number,
        public marcador :boolean,
        public observar :boolean,
        public marcaId :number,
        public zonaPrecioId :number,
        public marcaCodigo :string,
        public zonaPrecioCodigo: string,
        public marcaNombre: string,
        public zonaPrecioNombre: string,
        )
        {}
}
